Mathlib/Tactic/NthRewrite.lean

```diff
@@ -17,11 +17,68 @@ namespace Mathlib.Tactic
 
 open Lean Elab Tactic Meta Parser.Tactic
 
-/-- `nth_rewrite` is a variant of `rewrite` that only changes the nth occurrence of the expression
-to be rewritten.
+/-- `nth_rewrite` is a variant of `rewrite` that only changes the `n`ᵗʰ _occurrence_ of the
+expression to be rewritten. `nth_rewrite n [eq₁, eq₂,..., eqₘ]` will rewrite the `n`ᵗʰ _occurrence_
+of each of the `m` equalities `eqᵢ`in that order. Occurrences are counted beginning with `1` in
+order of precedence.
 
-Note: The occurrences are counted beginning with `1` and not `0`, this is different than in
-mathlib3. The translation will be handled by mathport. -/
+For example,
+```lean
+example (h : a = 1) : a + a + a + a + a = 5 := by
+  nth_rewrite 2 [h]
+/-
+a: ℕ
+h: a = 1
+⊢ a + 1 + a + a + a = 5
+-/
+```
+Notice that the second occurrence of `a` from the left has been rewritten by `nth_rewrite`.
+
+To understand the importance of order of precedence, consider the example below
+```lean
+example (a b c : Nat) : (a + b) + c = (b + a) + c := by
+  nth_rewrite 2 [Nat.add_comm] -- ⊢ (b + a) + c = (b + a) + c
+```
+Here, although the occurrence parameter is `2`, `(a + b)` is rewritten to `(b + a)`. This happens,
+because in order of precedence, the first occurrence of `_ + _` is the one that adds `a + b` to `c`.
+The occurrence in `a + b` counts as the second occurrence.
+
+If a term `t` is introduced by rewriting with `eqᵢ`, then this instance of `t` will be counted as an
+_occurrence_ of `t` for all subsequent rewrites of `t` with `eqⱼ` for `j > i`. This behaviour is
+illustrated by the example below
+```lean
+example (h : a = a + b) : a + a + a + a + a = 0 := by
+  nth_rewrite 3 [h, h]
+/-
+a b: ℕ
+h: a = a + b
+⊢ a + a + (a + b + b) + a + a = 0
+-/
+```
+Here, the first `nth_rewrite` with `h` introduces an additional occurrence of `a` in the goal.
+That is, the goal state after the first rewrite looks like below
+```lean
+/-
+a b: ℕ
+h: a = a + b
+⊢ a + a + (a + b) + a + a = 0
+-/
+```
+This new instance of `a` also turns out to be the third _occurrence_ of `a`.  Therefore,
+the next `nth_rewrite` with `h` rewrites this `a`.
+```lean
+/-
+a b: ℕ
+h: a = a + b
+⊢ a + a + (a + b) + a + a = 0
```

Comment:
```suggestion
⊢ a + a + (a + b + b) + a + a = 0
```
Isn't it?

Reply:
No this is the proof state after the first rewrite.